src/Commands/NotificationQueuer.php

```diff
@@ -92,6 +100,57 @@ public function listen($media_type_id) {
     $this->io()->progressFinish();
   }
 
+  /**
+   * Removes notifications that are older than the entities that reference them.
+   *
+   * @param \Lullabot\Mpx\DataService\Notification[] $notifications
+   *   An array of notifications.
+   * @param string $media_type_id
+   *   The media type being imported.
+   *
+   * @return array
+   *   The filtered array.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   *   The only way this could happen is if the media module was missing
+   *   (thrown from getStorage call).
+   */
+  private function filterByDate(array $notifications, string $media_type_id): array {
+    /* @todo I borrowed this search code from the DataObjectImporter
+     * maybe there is a better place to publicly store this functionality. */
+
+    /** @var \Drupal\Media\MediaStorage $media_storage */
+    $media_storage = \Drupal::entityManager()->getStorage('media');
+    /** @var \Drupal\Media\Entity\MediaType $media_type */
+    $media_type = MediaType::load($media_type_id);
+    $source = $media_type->getSource();
+    $source_field = $source->getSourceFieldDefinition($media_type)->getName();
+
+    $notifications = array_filter($notifications, function (MpxNotification $notification) use ($media_storage, $source_field) {
+      $notificationDate = $notification->getEntry()->getUpdated()->format("U");
```

Comment:
Do we need to update `mpx-php` as well for the getUpdated method on ObjectInterface?

Reply:
Ya that would make sense. Looking at the the interface compared to all the implemented functions on the classes, it looks like there is a lot that could be added to the interface. Did you want my to make a ticket to have the interface updated with all the common and useful public functions? And if so, do you want that to be pre-beta work?

Reply:
Yes, let's fix it. Otherwise we have the potential for FATALs down the line:

<img width="793" alt="notificationqueuer php drupal8 2018-10-25 10-22-55" src="https://user-images.githubusercontent.com/255023/47507340-06e55700-d840-11e8-9f5a-1d0398db0d5c.png">

I would say certainly add updated, and do a quick look for anything else I've missed. It doesn't need to be very in-depth.